### PR TITLE
[core] OOM killer: adjust defaults

### PR DESCRIPTION
**Pull request by clarng**
## Why are these changes needed?

Simplify OOM killer to use only fractional memory threshold by default.

Clean up nightly test to use the defaults only - no overrides. 

Updated docs and also include the exponential backoff behavior.

## Related issue number

https://github.com/ray-project/ray/issues/30433

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by clarng** [comment]
Release test run: https://buildkite.com/ray-project/release-tests-pr/builds/22147

File: release/ray_release/cluster_manager/cluster_manager.py [hunk 1]
```diff
@@ -55,6 +55,11 @@ def set_cluster_env(self, cluster_env: Dict[str, Any]):
         self.cluster_env["env_vars"][
             "RAY_memory_monitor_interval_ms"
         ] = self.cluster_env["env_vars"].get("RAY_memory_monitor_interval_ms", "250")
+        self.cluster_env["env_vars"][
+            "RAY_memory_usage_threshold_fraction"
+        ] = self.cluster_env["env_vars"].get(
+            "RAY_memory_usage_threshold_fraction", "0.95"
```
Comment (rkooo567) [inline release/ray_release/cluster_manager/cluster_manager.py:61]: maybe we should remove it?

File: src/ray/common/ray_config_def.h [hunk 2]
```diff
@@ -97,7 +97,7 @@ RAY_CONFIG(uint64_t, memory_monitor_interval_ms, 250)
 /// This value is useful for larger host where the memory_usage_threshold_fraction could
 /// represent a large chunk of memory, e.g. a host with 64GB of memory and 0.9 threshold
 /// means 6.4 GB of the memory will not be usable.
-RAY_CONFIG(int64_t, min_memory_free_bytes, (int64_t)512 * 1024 * 1024)
+RAY_CONFIG(int64_t, min_memory_free_bytes, (int64_t)-1)
```
Comment (rkooo567) [inline src/ray/common/ray_config_def.h:100]: can you comment what -1 means?

File: src/ray/common/ray_config_def.h [hunk 3]
```diff
@@ -84,7 +84,7 @@ RAY_CONFIG(uint64_t, raylet_check_gc_period_milliseconds, 100)
 /// memory_usage_threshold_fraction and free space is below the min_memory_free_bytes then
 /// it will start killing processes to free up the space.
 /// Ranging from [0, 1]
-RAY_CONFIG(float, memory_usage_threshold_fraction, 0.98)
+RAY_CONFIG(float, memory_usage_threshold_fraction, 0.95)
```
Comment (rkooo567) [inline src/ray/common/ray_config_def.h:87]: why did we change this ti 0.95 from 0.98?

**Review by rkooo567 — APPROVED** [review]
Love the doc became way shorter!

File: release/ray_release/cluster_manager/cluster_manager.py (hunk 1 above)
Comment (clarng) [reply release/ray_release/cluster_manager/cluster_manager.py:61]: For now we can't , product is override this to 1. going to ping @allenyin55 so we can remove this override

File: src/ray/common/ray_config_def.h (hunk 3 above)
Comment (clarng) [reply src/ray/common/ray_config_def.h:87]: To give more room chance to kill the process before it becomes worker crash, the original threshold for the legacy oom killer is also .95

**Comment by clarng** [comment]
looks ready for merge (PR nightly not worse than master) : https://buildkite.com/ray-project/release-tests-pr/builds/22147

**Review by ericl — APPROVED** [review]

**Pull request merged**